### ISSUE TITLE
[Bug] prefer_ipv6 not working as expected

### ISSUE THREAD

**Issue by cross** [Chia-Network/chia-blockchain#15192]
### What happened?

NB:  I worked with others to add IPv6 name support via #8552 (via #8861).

I have updated a host formerly running chia 1.3.4 to 1.7.1.  I've modified the `config.yaml` to list the farmer hostname, and set `prefer_ipv6` to True.  With 1.3.4, this caused the farmer to receive a connection from the IPv6 address of this host.  Now, it receives a connection via IPv4.

The networking hasn't changed on either host.

### Version

1.7.1

### What platform are you using?

Linux

### What ui mode are you using?

CLI

### Relevant log output

```shell
chia farm summary (when using 1.3.4 harvester):
Remote Harvester for IP: 2123:1234:5678:90ab::28
   24 plots of size: 6.837 TiB

chia farm summary (when using 1.7.1 harvester):
Remote Harvester for IP: 10.10.10.28
   24 plots of size: 6.837 TiB
```

**Comment by cross** [comment]
Additional useful note:  It's a name-mapping thing, not an IPv6 connectivity thing.  If I switch the specification of `farmer_peer` to the raw IPv6 address, it connects as it used to.  But, neither DNS nor any networking on either host has changed, so it's something in chia that's changed.

**Comment by cross** [comment]
Follow-up:  I installed `chia-blockchain-cli_1.8.0` and the same problem does not exist.  The hostname is resolved to its IPv6 address for the connection to the farmer.
You may close this if you like, unless you want to track and fix it in a 1.7.x release.

**Comment by emlowe** [comment]
Closing as fixed in 1.8.0
